compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -487,6 +489,9 @@ private BLangConstantValue calculateMod(BLangConstantValue lhs, BLangConstantVal
     }
 
     private Object calculateNegationForInt(BLangConstantValue value) {
+        if ((Long) (value.value) == Long.MIN_VALUE) {
```

Comment:
Do we have a test to cover this change?

Reply:
Yes https://github.com/ballerina-platform/ballerina-lang/pull/37726/files#diff-f7b80acafb6a5c9b42be1aea1acfa78b59911c59f6a479ac5cc5265b47e1a213R336